pkg/util/etcdutil/migrate.go

```diff
@@ -119,16 +121,20 @@ func createMigratedEtcdCluster(httpcli *http.Client, host, podIP string) error {
   }
 }`, podIP))
 
-	resp, err := httpcli.Post(
-		fmt.Sprintf("%s/apis/coreos.com/v1/namespaces/kube-system/etcdclusters", host),
-		"application/json", bytes.NewReader(b))
-	if err != nil {
-		return err
+	uri := "/apis/coreos.com/v1/namespaces/kube-system/etcdclusters"
+	res := restclient.Post().RequestURI(uri).Body(b).Do()
+	// Where do i put "application/json" ?
+	//fmt.Sprintf("%s/apis/coreos.com/v1/namespaces/kube-system/etcdclusters", host),"application/json", bytes.NewReader(b))
```

Comment:
Yaml may not work at all at the moment: https://github.com/kubernetes/kubernetes/issues/37455 -- but maybe this instance was? Or does this convert the body for us to json?

Otherwise, you should be able to add `SetHeader` to the chain:

`restclient.Post().RequestURI(uri).SetHeader("content-type", "application/yaml").Body(b).Do()`

If it's still causing problems, may need to temporarily change it to json until the issue is resolved.